#! /bin/sh /usr/share/dpatch/dpatch-run

## 10_dd_replay_option.dpatch by Thomas Schmidt <tschmidt@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Allowes to choose if to record and to replay Dolby Digital 
## DP: independently (to work around some audio issues with the 
## DP: dvd-plugin)

@DPATCH@
diff -urNad vdr-1.3.32/config.c /tmp/dpep.7sbpO9/vdr-1.3.32/config.c
--- vdr-1.3.32/config.c	2005-09-11 18:11:45.024178024 +0200
+++ /tmp/dpep.7sbpO9/vdr-1.3.32/config.c	2005-09-11 18:11:45.795060832 +0200
@@ -365,6 +365,7 @@
   VideoDisplayFormat = 1;
   VideoFormat = 0;
   UpdateChannels = 4;
+  RecordDolbyDigital = 1;
   UseDolbyDigital = 1;
   ChannelInfoPos = 0;
   ChannelInfoTime = 5;
@@ -523,6 +524,7 @@
   else if (!strcasecmp(Name, "VideoDisplayFormat"))  VideoDisplayFormat = atoi(Value);
   else if (!strcasecmp(Name, "VideoFormat"))         VideoFormat        = atoi(Value);
   else if (!strcasecmp(Name, "UpdateChannels"))      UpdateChannels     = atoi(Value);
+  else if (!strcasecmp(Name, "RecordDolbyDigital"))  RecordDolbyDigital = atoi(Value);
   else if (!strcasecmp(Name, "UseDolbyDigital"))     UseDolbyDigital    = atoi(Value);
   else if (!strcasecmp(Name, "ChannelInfoPos"))      ChannelInfoPos     = atoi(Value);
   else if (!strcasecmp(Name, "ChannelInfoTime"))     ChannelInfoTime    = atoi(Value);
@@ -588,6 +590,7 @@
   Store("VideoDisplayFormat", VideoDisplayFormat);
   Store("VideoFormat",        VideoFormat);
   Store("UpdateChannels",     UpdateChannels);
+  Store("RecordDolbyDigital", RecordDolbyDigital);
   Store("UseDolbyDigital",    UseDolbyDigital);
   Store("ChannelInfoPos",     ChannelInfoPos);
   Store("ChannelInfoTime",    ChannelInfoTime);
diff -urNad vdr-1.3.32/config.h /tmp/dpep.7sbpO9/vdr-1.3.32/config.h
--- vdr-1.3.32/config.h	2005-09-11 18:11:45.025177872 +0200
+++ /tmp/dpep.7sbpO9/vdr-1.3.32/config.h	2005-09-11 18:11:45.795060832 +0200
@@ -257,6 +257,7 @@
   int VideoDisplayFormat;
   int VideoFormat;
   int UpdateChannels;
+  int RecordDolbyDigital;
   int UseDolbyDigital;
   int ChannelInfoPos;
   int ChannelInfoTime;
diff -urNad vdr-1.3.32/i18n.c /tmp/dpep.7sbpO9/vdr-1.3.32/i18n.c
--- vdr-1.3.32/i18n.c	2005-09-11 18:02:41.692776952 +0200
+++ /tmp/dpep.7sbpO9/vdr-1.3.32/i18n.c	2005-09-11 18:11:45.799060224 +0200
@@ -3262,8 +3262,29 @@
     "TV külgsuhe",
     "Video format",
   },
+  { "Setup.DVB$Record Dolby Digital",
+    "Dolby Digital Ton aufzeichnen",
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+    "",//TODO
+  },
   { "Setup.DVB$Use Dolby Digital",
-    "Dolby Digital Ton benutzen",
+    "Dolby Digital Ton wiedergeben",
     "Uporabljaj Dolby Digital",
     "",//TODO
     "Dolby Digital gebruiken",
diff -urNad vdr-1.3.32/menu.c /tmp/dpep.7sbpO9/vdr-1.3.32/menu.c
--- vdr-1.3.32/menu.c	2005-09-11 18:11:45.266141240 +0200
+++ /tmp/dpep.7sbpO9/vdr-1.3.32/menu.c	2005-09-11 18:11:45.801059920 +0200
@@ -2039,7 +2039,8 @@
   Add(new cMenuEditBoolItem(tr("Setup.DVB$Video format"),          &data.VideoFormat, "4:3", "16:9"));
   if (data.VideoFormat == 0)
      Add(new cMenuEditStraItem(tr("Setup.DVB$Video display format"), &data.VideoDisplayFormat, 3, videoDisplayFormatTexts));
-  Add(new cMenuEditBoolItem(tr("Setup.DVB$Use Dolby Digital"),     &data.UseDolbyDigital));
+  Add(new cMenuEditBoolItem(tr("Setup.DVB$Record Dolby Digital"),  &data.RecordDolbyDigital));
+  Add(new cMenuEditBoolItem(tr("Setup.DVB$Replay Dolby Digital"),  &data.UseDolbyDigital));
   Add(new cMenuEditStraItem(tr("Setup.DVB$Update channels"),       &data.UpdateChannels, 5, updateChannelsTexts));
   Add(new cMenuEditIntItem( tr("Setup.DVB$Audio languages"),       &numAudioLanguages, 0, I18nNumLanguages));
   for (int i = 0; i < numAudioLanguages; i++)
diff -urNad vdr-1.3.32/recorder.c /tmp/dpep.7sbpO9/vdr-1.3.32/recorder.c
--- vdr-1.3.32/recorder.c	2005-09-11 18:02:41.696776344 +0200
+++ /tmp/dpep.7sbpO9/vdr-1.3.32/recorder.c	2005-09-11 18:11:45.802059768 +0200
@@ -122,7 +122,7 @@
 }
 
 cRecorder::cRecorder(const char *FileName, int Ca, int Priority, int VPid, const int *APids, const int *DPids, const int *SPids)
-:cReceiver(Ca, Priority, VPid, APids, Setup.UseDolbyDigital ? DPids : NULL, SPids)
+:cReceiver(Ca, Priority, VPid, APids, Setup.RecordDolbyDigital ? DPids : NULL, SPids)
 ,cThread("recording")
 {
   // Make sure the disk is up and running:
@@ -131,7 +131,7 @@
 
   ringBuffer = new cRingBufferLinear(RECORDERBUFSIZE, TS_SIZE * 2, true, "Recorder");
   ringBuffer->SetTimeouts(0, 100);
-  remux = new cRemux(VPid, APids, Setup.UseDolbyDigital ? DPids : NULL, SPids, true);
+  remux = new cRemux(VPid, APids, Setup.RecordDolbyDigital ? DPids : NULL, SPids, true);
   writer = new cFileWriter(FileName, remux);
 }
 
